common/src/test/scala/org/neo4j/spark/util/Neo4jUtilTest.scala

```diff
@@ -11,4 +11,43 @@ class Neo4jUtilTest {
     Neo4jUtil.closeSafety(null)
   }
 
+  @Test
+  def testCreateGdsConfigurationMap(): Unit = {
+    val actual = Neo4jUtil.createGdsConfigurationMap(Map(
+      "graphName" -> "foo",
+      "configuration.number" -> "1",
+      "configuration.string" -> "foo",
+      "configuration.list" -> "['a', 1]",
+      "configuration.map.key" -> "value",
+    ))
+    val expected: java.util.Map[String, Object] = Map(
+      "graphName" -> "foo",
+      "configuration" -> Map(
+        "number" -> 1,
+        "string" -> "foo",
+        "list" -> Seq("a", 1).toList.asJava,
+        "map" -> Map(
+          "key" -> "value"
+        ).asJava
+      ).asJava
+    ).asJava
+    Assert.assertEquals(expected, actual)
+
+    val ucActual = Neo4jUtil.createGdsConfigurationMap(Map(
+      "graphName" -> "myGraph",
+      "nodeProjection" -> "Website",
+      "relationshipProjection.LINK.indexInverse" -> "true",
```

Comment:
Ah that's really cool. You might wanna add a test for loading relationship properties, maybe even with a user-defined default value.